lib/features/neural/config.dart

```diff
@@ -60,11 +97,146 @@ class NeuralConfigState extends ConsumerState<NeuralConfig> {
+                String? hiddenNeuronsError =
+                    validateInteger(_hiddenNeuronsController.text, min: 1);
+                String? maxNWtsError =
+                    validateInteger(_maxNWtsController.text, min: 1);
+                String? maxitError =
+                    validateInteger(_maxitController.text, min: 1);
+
+                // Collect all errors.
+                List<String> errors = [
+                  if (hiddenNeuronsError != null)
+                    'Hidden Neurons: $hiddenNeuronsError',
+                  if (maxNWtsError != null) 'Max NWts: $maxNWtsError',
+                  if (maxitError != null) 'Maxit: $maxitError',
+                ];
+
+                // Check if there are any errors.
+                if (errors.isNotEmpty) {
+                  // Show a warning dialog if validation fails.
+                  showDialog(
+                    context: context,
+                    builder: (context) => AlertDialog(
+                      title: const Text('Validation Error'),
+                      content: Text(
+                        'Please ensure all input fields are valid before building the nnet model:\n\n${errors.join('\n')}',
+                      ),
+                      actions: [
+                        TextButton(
+                          onPressed: () {
+                            Navigator.of(context).pop();
+                          },
+                          child: const Text('OK'),
+                        ),
+                      ],
+                    ),
+                  );
+
+                  return;
+                } else {
+                  ref.read(nnetTraceProvider.notifier).state = nnetTrace;
+                  ref.read(nnetSkipProvider.notifier).state = nnetSkip;
+
+                  ref.read(hiddenNeuronsProvider.notifier).state =
+                      int.parse(_hiddenNeuronsController.text);
+                  ref.read(maxNWtsProvider.notifier).state =
+                      int.parse(_maxNWtsController.text);
+                  ref.read(maxitProvider.notifier).state =
+                      int.parse(_maxitController.text);
+
+                  // Run the R scripts.
+                  await rSource(context, ref, 'model_template');
+                  await rSource(context, ref, 'model_build_neural_net');
+                }
               },
               child: const Text('Build Neural Network'),
             ),
+
+            configWidgetSpace,
+
+            const Text(
+              'Trace',
+              style: normalTextStyle,
+            ),
+            Checkbox(
+              key: const Key('NNET Trace'),
+              value: nnetTrace,
+              onChanged: (value) {
+                setState(() {
+                  nnetTrace = value!;
+                  ref.read(nnetTraceProvider.notifier).state = value;
+                });
+              },
+            ),
+
+            configWidgetSpace,
+
+            const Text(
```

Comment:
Why is Trace capitalized and Skip is not? It is really important to be consistent.

Please also add a DelayedTooltip to each of these to explain the purpose. Thanks.